cmd/tailscale/cli/up.go

```diff
@@ -193,12 +193,38 @@ func calcAdvertiseRoutes(advertiseRoutes string, advertiseDefaultRoute bool) ([]
 	return routes, nil
 }
 
+// peerWithTailscaleIP returns the peer in st with the provided
+// Tailscale IP.
+func peerWithTailscaleIP(st *ipnstate.Status, ip netaddr.IP) (ps *ipnstate.PeerStatus, ok bool) {
+	for _, ps := range st.Peer {
+		for _, ip2 := range ps.TailscaleIPs {
+			if ip == ip2 {
+				return ps, true
+			}
+		}
+	}
+	return nil, false
+}
+
+// exitNodeIPOfArg maps from a user-provided CLI flag value to an IP
+// address they want to use as an exit node.
 func exitNodeIPOfArg(arg string, st *ipnstate.Status) (ip netaddr.IP, err error) {
 	if arg == "" {
 		return ip, errors.New("invalid use of exitNodeIPOfArg with empty string")
 	}
 	ip, err = netaddr.ParseIP(arg)
 	if err == nil {
+		// If we're online already and have a netmap, double check that the IP
+		// address specified is valid.
+		if st.BackendState == "Running" {
```

Comment:
Do we have to check for running in the other code path as well?

Reply:
Nope. Normally it's allowed to set an arbitrary IP address when you're not running and there's no netmap. The backend will lazily remap a Tailscale IP to a Node StableID.

So this check is saying "If we're Running and we know the state of the world, then check here to fail them early."

The check below is trying to get any IP at all to give to the backend, which won't accept a name. It doesn't matter what state we're in. If can map it in Stopped state, that's good enough.